From bbb633ce344537e66e286ee29c2632818c270fd4 Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan-RA5478 <Ranjani.Vaidyanathan@freescale.com>
Date: Mon, 2 Feb 2009 12:47:52 -0600
Subject: [PATCH] ENGR00106771-1: Fix DVFS core failure.

Added support for using ARM freq divider bits instead of changing PLL frequency.

Signed-off-by: Ranjani Vaidyanathan-RA5478 <Ranjani.Vaidyanathan@freescale.com>
---
 arch/arm/plat-mxc/cpufreq.c    |    4 +---
 include/asm-arm/arch-mxc/mxc.h |    1 +
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/arch/arm/plat-mxc/cpufreq.c b/arch/arm/plat-mxc/cpufreq.c
index dea89bf..184887e 100644
--- a/arch/arm/plat-mxc/cpufreq.c
+++ b/arch/arm/plat-mxc/cpufreq.c
@@ -79,10 +79,8 @@ static int set_cpu_freq(int freq)
 	int i;
 
 	org_cpu_rate = clk_get_rate(cpu_clk);
-	if (org_cpu_rate == freq)
-		return ret;
 
-	if (!axi_c_clk_support)
+	if (org_cpu_rate == freq)
 		return ret;
 
 	for (i = 0; i < cpu_wp_nr; i++) {
diff --git a/include/asm-arm/arch-mxc/mxc.h b/include/asm-arm/arch-mxc/mxc.h
index b2e120f..8e24073 100644
--- a/include/asm-arm/arch-mxc/mxc.h
+++ b/include/asm-arm/arch-mxc/mxc.h
@@ -394,6 +394,7 @@ struct cpu_wp {
 	u32 mfd;
 	u32 mfn;
 	u32 cpu_voltage;
+	u32 cpu_podf;
 };
 
 struct cpu_wp *get_cpu_wp(int *wp);
-- 
1.5.4.4

